modules/gdscript/gdscript_parser.cpp

```diff
@@ -7731,6 +7736,17 @@ void GDScriptParser::_check_function_types(FunctionNode *p_function) {
 		p_function->return_type.has_type = false;
 		p_function->return_type.may_yield = true;
 	}
+
+#ifdef DEBUG_ENABLED
+	for (Map<StringName, LocalVarNode *>::Element *E = p_function->body->variables.front(); E; E = E->next()) {
+		LocalVarNode *lv = E->get();
+		for (int i = 0; i < current_class->variables.size(); i++) {
+			if (current_class->variables[i].identifier == lv->name) {
+				_add_warning(GDScriptWarning::SHADOWED_VARIABLE, p_function->line, lv->name, itos(current_class->variables[i].line));
```

Comment:
I've done the changes, the only problem I have is with this line. How do I get the line of the variable. I don't see any `line` variable in LocalVarNode.

Reply:
You can use `lv->line`. Every parser node has line and column.

Reply:
I just realized a few minutes ago, for some reason my editor dont's show the ->line xD